### ISSUE TITLE
Recursive Arrays

### ISSUE THREAD

**Issue by YStrauch** [sandorfr/gulp-swagger-generator#38]
Hello,

if you specify a an array within an array as followed:

```yaml
recursive_array:
        type: array
        items:
          type: array
          items:
            type: string
```

then the library will crash. I found the issue in the typescriptTypeMapper (and a quick fix):
```js
          else if (type === 'array') {
                /* The error occures because asArray() is not set for recursive arrays */
                return this.getType(property.items).asArray();

                /* Fix: A simple check if asArray is defined */
                if (!this.getType(property.items).asArray) {
                    return this.getType(property.items);
                }
                return this.getType(property.items).asArray();
            }
```

Hope I didn't miss something ;)

**Comment by sandorfr** [comment]
Hi @YStrauch, thanks for reporting that issue. I'm afraid there is more to the story. Type system in the legacy version of the generator has a pretty bad design :( (sorry about that). I think your patch proposal will just hide the error but won't fix the core issue.

[Current templates](https://github.com/geeklearningio/gulp-swagger-generator/blob/legacy/lib/templates/typescript/angular/Type.hbs#L6) won't generate the appropriate type anyway as it's just appending a `[]` and won't trigger a recursive call to the type template.

This has been addressed in the `vnext` by updating the [Abstracted type model](https://github.com/geeklearningio/gl-swagger-generator/blob/develop/src/node-swagger-generator/lib/typing.ts#L118-L126) and [Language type model](https://github.com/geeklearningio/gl-swagger-generator/blob/develop/src/node-swagger-generator/lib/languages/typescript.ts#L222-L228) and the [Templates](https://github.com/geeklearningio/gl-swagger-generator/blob/develop/src/node-swagger-generator/templates/csharp/netstandard/1.3/_typeCore.hbs#L6)

My priority is actually to get this vnext documented, tested and released, so I won't have time to backport this to legacy branch.

**Comment by sandorfr** [comment]
FYI, we've released angular 1.5 templates for the new generator which doesn't have this issue.
